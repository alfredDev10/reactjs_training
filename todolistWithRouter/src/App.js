import React from 'react';
import './App.css';
import {BrowserRouter as Router, Link, Route} from 'react-router-dom';
import TodoList from './TodoList';
import Welcome from './Welcome';

function App() {
    return (
        <Router>
            <Link to="/todos">Go to my todos!</Link>
            <Route path="/" exact component={Welcome}></Route>
            <Route path="/todos" component={TodoList}></Route>
        </Router>
    );
}

export default App;
