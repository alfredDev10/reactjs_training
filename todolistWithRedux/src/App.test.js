import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('snapshot',()=>{
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
})