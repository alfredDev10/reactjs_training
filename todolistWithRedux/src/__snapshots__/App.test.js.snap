// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <ul>
    <li>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      Learn ReactJS
    </li>
    <li>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox"
      />
      Learn css
    </li>
    <li>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      Web development
    </li>
  </ul>,
  <label>
     New Task:
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>,
  <button
    onClick={[Function]}
  >
    Validate
  </button>,
]
`;
